build: correctly publish `angular-in-memory-web-api` to CI build artifacts

gkalpak opened the pull request:
Several fixes/improvements to building the `angular-in-memory-web-api` package and publishing it to CI build artifacts.
See individual commits for details.

mary-poppins commented:
You can preview 3db0e3c at https://pr41429-3db0e3c.ngbuilds.io/.

mary-poppins commented:
You can preview 7c5cf47 at https://pr41429-7c5cf47.ngbuilds.io/.

mary-poppins commented:
You can preview 548001a at https://pr41429-548001a.ngbuilds.io/.

josephperrott commented on scripts/build/dev-infra-builder.js:
```diff
@@ -8,29 +8,39 @@
 
+  // Create the output directory.
+  const absDestDir = resolve(baseDir, destDir);
+  if (!test('-d', absDestDir)) mkdir('-p', absDestDir);
```
```suggestion
  if (!test('-d', absDestDir)) {
    mkdir('-p', absDestDir);
  }
```

josephperrott reviewed (approved):
LGTM

Just one nit comment

gkalpak committed:
build: make the build scripts for the various packages consistent

This commit makes the build scripts for the various packages (framework,
`@angular/dev-infra-private`, `angular-in-memory-web-api`, `zone.js`)
consistent. This makes it easier to maintain them (e.g. make similar
changes across all build scripts).

gkalpak committed:
build: correctly publish `angular-in-memory-web-api` as CI build artifact

Previously, the `angular-in-memory-web-api` package was built in
`dist/packages-dist/misc/angular-in-memory-web-api/`. This was different
from other Angular packages, which were placed directly in
`dist/packages-dist/`. This caused the `create-package-archives.sh`
script to create an invalid `misc.tgz` archive (i.e. treating the
`misc/` subdirectory as a package).
See, for example, the artifacts [here][1].

This commit changes the build scripts to have the
`angular-in-memory-web-api` package built in
`dist/angular-in-memory-web-api-dist/`, similar to how the `zone.js`
package is handled. It also updates the CircleCI config to correctly
publish the `angular-in-memory-web-api` package to CI build artifacts.

[1]: https://circleci.com/gh/angular/angular/951491

gkalpak committed:
build(docs-infra): enable linting for `ng-packages-installer` scripts

This commit enables linting for the scripts in
`aio/tools/ng-packages-installer/`. It also makes the necessary changes
to the files to make linting pass.

gkalpak committed:
build: build `angular-in-memory-web-api` when building packages with Ivy

This commit updates the `build-ivy-npm-packages.js` script to also build
the `angular-in-memory-web-api` package. This could be used if someone
wanted to test the Ivy packages in docs examples or an integration test.

gkalpak committed:
build(docs-infra): update `eslint` and `eslint-plugin-jasmine` to latest versions

This commit updates the `eslint` and `eslint-plugin-jasmine` packages to
latest versions to take advantage of latest fixes and improvements.

gkalpak committed:
fixup! build: make the build scripts for the various packages consistent

mary-poppins commented:
You can preview 3d5a2c1 at https://pr41429-3d5a2c1.ngbuilds.io/.
